src/validation.cpp

```diff
@@ -2075,15 +2076,18 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
 
     CBlockUndo blockundo;
 
+    // Precomputed transaction data pointers must not be invalidated
+    // until after control has run the script checks (potentially
+    // in multiple threads). Keep txsdata in scope for as long as
+    // control.
```

Comment:
Suggestion here or for a follow-up: write and use the word `control` here in a way that indicates more clearly that it's not just the word "control".

Reply:
Added backticks to it's more obviously referring to the code.

Reply:
That could work, but it'd need something like a shared_ptr (there is one PrecomputedTransactionData per *transaction*, but a CScriptCheck per *txin*). So I think that's perhaps an unreasonable overhead.

Reply:
@sipa Was this suppose to be in response to https://github.com/bitcoin/bitcoin/pull/18401#discussion_r401896789?

Reply:
Oops, yes.